griptape/utils/dict_utils.py

```diff
@@ -55,3 +55,12 @@ def dict_merge(dct: Optional[dict], merge_dct: Optional[dict], *, add_keys: bool
             dct[key] = merge_dct[key]
 
     return dct
+
+
+def dict_merge_opt(dct: Optional[dict], merge_dct: Optional[dict], *, add_keys: bool = True) -> Optional[dict]:
+    if dct is None:
+        return merge_dct
+    if merge_dct is None:
+        return dct
+
+    return dict_merge(dct, merge_dct, add_keys=add_keys)
```

Comment:
Why does this method need to exist? Can its functionality be merged into `dict_merge`?

Reply:
every instance of its usage would need an updated type hint. tried to be unobtrusive

Reply:
deleted